### PR TITLE
chore: linters now work correctly with Windows Subsystem for Linux (WSL)

### LINKED ISSUE #55: Repo build and dev env doesn’t work on Windows 10
pkglint doesn't work on Windows it seems - or something is running chmod which obviously does not exist.

```
C:\Users\lpack\Documents\Code\aws-cdk>node --version
v8.11.2
```

```
lerna info lifecycle prepare
lerna sill runScript prepare pkglint
lerna sill runScriptInDir prepare [] pkglint
lerna sill getExecOpts { cwd: 'C:\\Users\\lpack\\Documents\\Code\\aws-cdk\\packages\\pkglint' }
lerna sill pkglint
lerna ERR! execute Error occured with 'pkglint' while running 'npm run prepare'
lerna ERR! npm run prepare [pkglint] Output from stdout:

> pkglint@0.6.0 prepare C:\Users\lpack\Documents\Code\aws-cdk\packages\pkglint
> tsc && tslint -p . && chmod +x bin/pkglint


lerna ERR! npm run prepare [pkglint] Output from stderr:
'chmod' is not recognized as an internal or external command,
operable program or batch file.
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! pkglint@0.6.0 prepare: `tsc && tslint -p . && chmod +x bin/pkglint`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the pkglint@0.6.0 prepare script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\lpack\AppData\Roaming\npm-cache\_logs\2018-06-06T21_16_29_706Z-debug.log

lerna ERR! execute Error occured with 'pkglint' while running 'npm run prepare'
{ Error: Command failed: npm run prepare
'chmod' is not recognized as an internal or external command,
operable program or batch file.
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! pkglint@0.6.0 prepare: `tsc && tslint -p . && chmod +x bin/pkglint`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the pkglint@0.6.0 prepare script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\lpack\AppData\Roaming\npm-cache\_logs\2018-06-06T21_16_29_706Z-debug.log

> pkglint@0.6.0 prepare C:\Users\lpack\Documents\Code\aws-cdk\packages\pkglint
> tsc && tslint -p . && chmod +x bin/pkglint


    at Promise.all.then.arr (C:\Users\lpack\Documents\Code\aws-cdk\node_modules\lerna\node_modules\execa\index.js:236:11)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)
  code: 1,
  killed: false,
  stdout: '\n> pkglint@0.6.0 prepare C:\\Users\\lpack\\Documents\\Code\\aws-cdk\\packages\\pkglint\n> tsc && tslint -p . && chmod +x bin/pkglint\n\n',
  stderr: '\'chmod\' is not recognized as an internal or external command,\r\noperable program or batch file.\r\nnpm ERR! code ELIFECYCLE\nnpm ERR! errno 1\nnpm ERR! pkglint@0.6.0 prepare: `tsc && tslint -p . && chmod +x bin/pkglint`\nnpm ERR! Exit status 1\nnpm ERR! \nnpm ERR! Failed at the pkglint@0.6.0 prepare script.\nnpm ERR! This is probably not a problem with npm. There is likely additional logging output above.\n\nnpm ERR! A complete log of this run can be found in:\nnpm ERR!     C:\\Users\\lpack\\AppData\\Roaming\\npm-cache\\_logs\\2018-06-06T21_16_29_706Z-debug.log\n',
  failed: true,
  signal: null,
  cmd: 'npm run prepare',
  timedOut: false,
  pkg:
   Package {
     _package:
      { name: 'pkglint',
        version: '0.6.0',
        private: true,
        description: 'Validate and fix package.json files',
        main: 'lib/index.js',
        types: 'lib/index.d.ts',
        repository: [Object],
        pkglint: [Object],
        bin: [Object],
        scripts: [Object],
        keywords: [Array],
        author: [Object],
        license: 'Apache-2.0',
        devDependencies: [Object],
        dependencies: [Object] },
     _location: 'C:\\Users\\lpack\\Documents\\Code\\aws-cdk\\packages\\pkglint' },
  exitCode: 1 }

C:\Users\lpack\Documents\Code\aws-cdk>node --version
v8.11.2

C:\Users\lpack\Documents\Code\aws-cdk>
```

### PR DESCRIPTION
**Pull request by Litee**
Small changes to make builds work in Windows Subsystem for Linux (WSL). Verified by **partial** building in Windows 10 + Ubuntu 18.04 WSL.

fixes #55

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by Litee** [comment]
@nija-at Any chance you can confirm my good intentions for the first-time contribution? :)

File: tools/@aws-cdk/eslint-plugin/lib/rules/invalid-cfn-imports.ts [hunk 1]
```diff
@@ -143,6 +143,7 @@ function getCdkRootDir(filename: string): string | undefined {
 }
 
 function isAlphaPackage(packageDir: string): boolean {
+  console.log(`Checking ${packageDir}`);
```
Comment (nija-at) [inline tools/@aws-cdk/eslint-plugin/lib/rules/invalid-cfn-imports.ts:146]: revert

File: tools/@aws-cdk/pkglint/lib/rules.ts [hunk 2]
```diff
@@ -118,8 +118,10 @@ export class RepositoryCorrect extends ValidationRule {
   public validate(pkg: PackageJson): void {
     expectJSON(this.name, pkg, 'repository.type', 'git');
     expectJSON(this.name, pkg, 'repository.url', 'https://github.com/aws/aws-cdk.git');
-    const pkgDir = path.relative(monoRepoRoot(), pkg.packageRoot);
-    expectJSON(this.name, pkg, 'repository.directory', pkgDir);
+    const osSpecificRepositoryDirectory = path.relative(monoRepoRoot(), pkg.packageRoot);
+    // Enforcing '/' as separator for builds to work in Windows.
+    const posixStyleRepositoryDirectory = osSpecificRepositoryDirectory.split(path.sep).join('/');
+    expectJSON(this.name, pkg, 'repository.directory', posixStyleRepositoryDirectory);
```
Comment (nija-at) [inline tools/@aws-cdk/pkglint/lib/rules.ts:124]: keep the terse names.

```suggestion
    const pkgDir = path.relative(monoRepoRoot(), pkg.packageRoot);
    // Enforcing '/' as separator for builds to work in Windows.
    const osPkgDir = pkgDir.split(path.sep).join('/');
    expectJSON(this.name, pkg, 'repository.directory', osPkgDir);
```

**Review by nija-at — DISMISSED** [review]

**Comment by Litee** [comment]
I messed up experimenting with Gitpod, so had to kill the branch. New PR: #17513

**Pull request closed without merging**